src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +36,10 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user)
+                .filter(u -> u.getAge() != null)
+                .filter(u -> u.getAge() >= 30 && u.getAge() <= 45)
+                .isPresent();
```

Comment:
```suggestion
        return Optional.ofNullable(user)
                .map(User::getAge)
                .filter(age-> age >= 30 && age <= 45)
                .isPresent();
```
이런식으로도 가능합니다 getAge가 null인 경우 filter에 순회 대상에서 빠지기 때문이죠